velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -70,33 +123,86 @@ std::unique_ptr<velox::dwrf::Writer> HiveDataSink::createWriter() {
   // Without explicitly setting flush policy, the default memory based flush
   // policy is used.
 
-  auto hiveWriterParameters =
-      std::dynamic_pointer_cast<const HiveWriterParameters>(
-          writeProtocol_->getWriterParameters(
-              insertTableHandle_, connectorQueryCtx_));
-  VELOX_CHECK_NOT_NULL(
-      hiveWriterParameters,
-      "Hive data sink expects write parameters for Hive.");
-  writerParameters_.emplace_back(hiveWriterParameters);
-
-  auto writePath = fs::path(hiveWriterParameters->writeDirectory()) /
-      hiveWriterParameters->writeFileName();
+  auto writePath = fs::path(writerParameters->writeDirectory()) /
+      writerParameters->writeFileName();
   auto sink = dwio::common::DataSink::create(writePath);
   return std::make_unique<Writer>(
       options, std::move(sink), *connectorQueryCtx_->memoryPool());
 }
 
+std::shared_ptr<const HiveWriterParameters> HiveDataSink::getWriterParameters(
+    const std::optional<std::string>& partition) const {
+  auto hiveTableWriteHandle =
+      std::dynamic_pointer_cast<const HiveInsertTableHandle>(
+          insertTableHandle_);
+  VELOX_CHECK_NOT_NULL(
+      hiveTableWriteHandle, "Hive connector expects Hive insert table handle.");
+
+  std::string targetFileName;
+  std::string writeFileName;
+  switch (commitStrategy_) {
+    case CommitStrategy::kNoCommit: {
+      targetFileName = fmt::format(
+          "{}_{}_{}",
+          connectorQueryCtx_->taskId(),
+          connectorQueryCtx_->driverId(),
+          makeUuid());
+      writeFileName = targetFileName;
+      break;
+    }
+    case CommitStrategy::kTaskCommit: {
+      targetFileName = fmt::format(
+          "{}_{}_{}",
+          connectorQueryCtx_->taskId(),
+          connectorQueryCtx_->driverId(),
+          0);
+      writeFileName =
+          fmt::format(".tmp.velox.{}_{}", targetFileName, makeUuid());
+      break;
+    }
+    default:
+      VELOX_UNREACHABLE();
+  }
+
+  HiveWriterParameters::UpdateMode updateMode;
+  if (hiveTableWriteHandle->isInsertTable()) {
```

Comment:
nit: this method is rather long; consider extracting this logic into a helper method

```
auto updateMode = getUpdateMode();
```

Reply:
Helper func added